src/MongoDB.Driver.GridFS/GridFSForwardOnlyUploadStream.cs

```diff
@@ -76,7 +78,8 @@ internal class GridFSForwardOnlyUploadStream<TFileId> : GridFSUploadStream<TFile
             _batchSize = batchSize;
 
             _batch = new List<byte[]>();
-            _md5 = IncrementalMD5.Create();
+            _md5 = disableMD5 ? null : IncrementalMD5.Create();
```

Comment:
CreateWriteBatchChunkDocuments in this same file accesses _md5 without checking if it is null.

Not sure why that didn't fail any tests with a NullReferenceException.

p.s. I think it's because you tested disableMD5 with a zero length file.

Reply:
Thank you, fixed!